File: src/main/java/com/christophsens/cvenhancer/prompt/ImprovePrompt.java
package com.christophsens.cvenhancer.prompt;

import org.springframework.ai.chat.prompt.Prompt;
import org.springframework.ai.chat.prompt.PromptTemplate;

public class ImprovePrompt {
    private static final String TEMPLATE = """
        Improve this cv: "{cv}".
        Your response must be in the language of the CV.
        """;

    private ImprovePrompt() {
    }

    public static Prompt getPrompt(String cv) {
        PromptTemplate promptTemplate = new PromptTemplate(TEMPLATE);
        promptTemplate.add("cv", cv);
        return promptTemplate.create();
    }
}

File: src/main/kotlin/com/christophsens/cvenhancer/prompt/ImprovePrompt.kt
package com.christophsens.cvenhancer.prompt

import org.springframework.ai.chat.prompt.Prompt
import org.springframework.ai.chat.prompt.PromptTemplate

object ImprovePrompt {

    private val TEMPLATE = """
        Improve this cv: "{cv}".
        Your response must be in the language of the CV.
    """.trimIndent()

    fun getPrompt(cv: String): Prompt {
        val promptTemplate = PromptTemplate(TEMPLATE)
        promptTemplate.add("cv", cv)
        return promptTemplate.create()
    }
}